sapi/fpm/tests/gh12385.phpt

```diff
@@ -0,0 +1,46 @@
+--TEST--
+GH-12385 (flush with fastcgi does not force headers to be sent)
+--SKIPIF--
+<?php
+include "skipif.inc";
+?>
+--FILE--
+<?php
+
+require_once "tester.inc";
+
+$cfg = <<<EOT
+[global]
+error_log = {{FILE:LOG}}
+[unconfined]
+listen = {{ADDR}}
+pm = static
+pm.max_children = 1
+EOT;
+
+$code = <<<PHP
+<?php
+header("X-Test: 12345");
+flush();
+var_dump(headers_sent());
+PHP;
+
+$tester = new FPM\Tester($cfg, $code);
+$tester->start();
+$tester->expectLogStartNotices();
+$response = $tester->request();
+$response->expectHeader("X-Test", "12345");
+$response->expectBody("bool(true)");
```

Comment:
```suggestion
$headersSentBeforeFlush = headers_sent();
flush();
$headersSentAfterFlush = headers_sent();
var_dump($headersSentBeforeFlush);
var_dump($headersSentAfterFlush);
PHP;

$tester = new FPM\Tester($cfg, $code);
$tester->start();
$tester->expectLogStartNotices();
$response = $tester->request();
$response->expectHeader("X-Test", "12345");
$response->expectBody("bool(false)\nbool(true)");
```

Reply:
I don't this this is necessary. We don't need to test what's given which is that headers are not sent at the beginning of request.